ccm_web/server/src/api/api.service.ts

```diff
@@ -139,8 +139,8 @@ export class APIService {
     let userRoles: CanvasRole[] = []
     try {
       userRoles = roleStringsToEnums(session.data.course.roles)
-    } catch (e) {
-      throw Error(`Role error for user "${user.loginId}": ${String(e.message)}`)
+    } catch (e: any) {
```

Comment:
I will differ to @ssciolla on this using `any` as type is something is followed as pattern in the code.

Reply:
I'm not sure I understand what you mean.  However, I can explain my choice of `any` there, if that helps.

I had seen the error message you reported earlier…

```
ccm_web     | server/src/api/api.service.ts(142,68): error TS2571: Object is of type 'unknown'.
```

So, I tried to specify the specific type of exception to be caught…

``` ts
    } catch (e: Error) {
```

That gave this error…

```
ccm_web  | server/src/api/api.service.ts(143,17): error TS1196: Catch clause variable type annotation must be 'any' or 'unknown' if specified.
```

Since `any` and `unknown` are the only types allowed and `unknown` gave an error, I decided to use `any`.  When I did that, the original error I saw (same as you reported) was resolved.

Reply:
The problem with `any` is that it's going to let you do anything to `e` (i.e. accessing `message` even though it doesn't exist, which could throw a runtime error). Looks like are casting `e.message` as `String`, so maybe it would be the string `undefined`. In general I've found it's better (and recommended) to avoid using `any` (though there are cases where it can be useful). 

You can check that `e` is an `Error` I believe, something like this:
```
  } catch (error: unknown) {
    if (error instanceof Error) {
      throw new Error(error.message)
    }
    throw new Error(String(error))
```